go_metrics/server.py

```diff
@@ -17,7 +21,34 @@ def parse_qs(qs):
         for (k, v) in _parse_qs(qs).iteritems())
 
 
+def HTTPBasic(method):
+    @functools.wraps(method)
+    def wrapper(self, *args, **kwargs):
+        backend = self.model.backend
+        config = backend.config
+        if not (getattr(config, 'basicauth_username', None) and
+                getattr(config, 'basicauth_password', None)):
+            return method(self, *args, **kwargs)
+
+        msg = "Authentication Required"
+        if "Authorization" in self.request.headers:
+            auth_type, data = self.request.headers["Authorization"].split()
+            if auth_type == "Basic":
+                usr, pwd = base64.b64decode(data).split(":", 1)
```

Comment:
With this line, and the `split` on line 35, what would happen if someone sends us a bad header (Would we give a 500 back)? Maybe we should catch exceptions in the splitting and b64decode, and send them an AuthenticationRequired if we cannot decode the header.